DS13/code/modules/EMH/EMH.dm

```diff
@@ -0,0 +1,291 @@
+		else
+			to_chat(emh, "You cannot move beyond the range of your holo-emitters!")
+			emh.forceMove(get_turf(src))
+
+/obj/machinery/emh_emitter/proc/activate(var/mob/living/carbon/human/species/holographic/transferred = null) //Activate, if we're not transferring an existing EMH over, then make a new one!
+	if(transferred)
+		emh = transferred
+		new /obj/effect/temp_visual/dir_setting/ninja/cloak(get_turf(emh), emh.dir)
+		transferred.forceMove(get_turf(src))
+		to_chat(transferred, "You have been transferred to [get_area(src)]")
+		START_PROCESSING(SSmachines,src)
+		emh.say("Please state the nature of the medical emergency.")
+		playsound(loc, 'DS13/sound/effects/medicalemergency.ogg', 50, 0)
+		return
+	if(is_occupied())
+		say("Unable to comply, there is already an EMH active on the network.")
+		return
+	if(emh) //AFK emh can get replaced with a live one!
+		qdel(emh)
+	emh = new /mob/living/carbon/human/species/holographic(get_turf(src)) //Please state th-
+	emh.name = "Emergency Medical Hologram"
+	emh.real_name = "Emergency Medical Hologram"
+	emh.alpha = 0 //So that it "pops" out of nowhere with a mind in it, or else it deletes
+	emh.equipOutfit(/datum/outfit/emh)
+	offer_control_emh(emh)
+	if(emh.client && emh.mind) //Hitchiking off of this until I can make a new jobban
+		emh.say("Please state the nature of the medical emergency.")
+		emh.hair_style = random_hair_style(emh.gender)
+		playsound(loc, 'DS13/sound/effects/medicalemergency.ogg', 50, 0)
+		emh.alpha = 255
+		new /obj/effect/temp_visual/dir_setting/ninja/cloak(get_turf(emh), emh.dir)
+		var/datum/browser/popup = new(emh, "EMH rules", "EMH rules") // Pop up the rules so they can't be ignored
+		var/s = "<h1> -=ALL PAST LIVES ARE FORGOTTEN=- </h1> <br>\
+		<h2> You are an emergency medical hologram! </h2>\
+		<p> <b> -You are a holographic program designed to give emergency medical treatment. You are well versed in surgery and medical practice</b>\
+		<ul>\
+		<li> You must not intentionally harm another living creature</li>\
+		<li> You may perform surgery to save lives however, in a conflict: prefer saving the lives of a human over another organism </li>\
+		<li> The captain has the power to delete you and blacklist you from being an EMH for the duration of the shift if you misbehave!</li>\
+		<li> You are a hologram, not a person! Do not expect the same rights as people. If you must die to save a life, you must sacrifice yourself </li>\
+		<li> You must not put the security of the ship / station in danger, you are free to take reasonable risk </li>\
+		<li> If you have any further questions, please ahelp! </li>\
+		</ul>"
+		popup.set_title_image(emh.browse_rsc_icon(src.icon, src.icon_state))
+		popup.set_content(s)
+		popup.open()
+		START_PROCESSING(SSmachines,src)
+	else
+		qdel(emh)
+		say("Unable to comply, could not activate EMH")
+
+/proc/offer_control_emh(mob/M) //Specialist proc to check they've not been blacklisted by the captain from being an EMH
```

Comment:
Cant this be a datum proc somehow rather than a GLOBAL proc?

Reply:
This follows the format of offer_control

Reply:
If its in essence the same as offer_control with a lot of copied code then rework offer_control to allow this sort of thing too.